Source/Kesmai.Server/Game/Entities/Mobiles/Creatures/CreatureEntity.BossProps.cs

```diff
@@ -0,0 +1,97 @@
+using System.IO;
+
+using Kesmai.Server.Accounting;
+using Kesmai.Server.Engines.Commands;
+using Kesmai.Server.Items;
+using Kesmai.Server.Miscellaneous;
+
+namespace Kesmai.Server.Game
+{
+    public class CreatureEntityBossProps
+    {
+        private CreatureEntity _creatureEntity;
+
+        public CreatureEntityBossProps(CreatureEntity creatureEntity)
+        {
+            _creatureEntity = creatureEntity;
+        }
+
+        /* This needs a look from Lux. 
+                I don't know if the implementation is right due to access of source code is limited. 
+                Overall this is to clean up the Bosses and to establish Default Props using a c# file. 
+
+                To what extent this could be used would be determined by the person implementing the bosses. 
+            */
+        public CreatureEntity SetBossProperties(BossType bossType, CreatureImmunity immunities = null, CreatureWeakness weakness = null)
+        {
+            if (immunities != null)
+            {
+                _creatureEntity.CreatureImmunity = immunities;
+            }
+
+            if (weakness != null)
+            {
+                _creatureEntity.CreatureWeakness = weakness;
+            }
+
+            switch(bossType)
+            {
+                case BossType.Minor:
```

Comment:
No default case means you're not going to assign anything if something beyond minor major or notable come through here. That's okay if that's the intent but generally default cases are assigned as a catch unexpected.

Reply:
no worries. Return null I guess or bail out.

Reply:
and done and back to you @Leolus